repos/system_upgrade/el7toel8/actors/checkkdegnome/libraries/library.py

```diff
@@ -0,0 +1,67 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledDesktopsFacts, InstalledKdeAppsFacts
+
+
+def check_kde_gnome():
+    desktopFacts = next(api.consume(InstalledDesktopsFacts))
+    kde_desktop_installed = desktopFacts.kde_installed
+    gnome_desktop_installed = desktopFacts.gnome_installed
+
+    # No desktop installed, we don't even care about apps as they are most likely not used or even installed
+    if not kde_desktop_installed and not gnome_desktop_installed:
+        api.current_logger().info("No desktop installed. Continuing with the upgrade.")
+        return
+
+    if kde_desktop_installed:
+        api.current_logger().info("KDE desktop is installed. Checking what we can do about it.")
+        if not gnome_desktop_installed:
+            api.current_logger().error("Cannot perform the upgrade because there is"
+                                       " no other desktop than KDE installed.")
+            # We cannot continue with the upgrade process
+            reporting.create_report([
+                reporting.Title("Cannot upgrade because there is no other desktop than KDE installed."),
+                reporting.Summary("With only KDE installed, there would be no other desktop env. after upgrade."),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Tags([
+                    reporting.Tags.UPGRADE_PROCESS
+                ]),
+                reporting.Flags([
+                    reporting.Flags.INHIBITOR
+                ]),
+                reporting.Remediation(
+                    hint="Install GNOME desktop to be able to upgrade.")
+                ])
+            return
+
+        # Assume both GNOME and KDE are installed in this state
+        api.current_logger().info("Upgrade can be performed, but KDE desktop will"
+                                  " be removed in favor of GNOME")
+        reporting.create_report([
+            reporting.Title("Upgrade can be performed, but KDE will be uninstalled."),
```

Comment:
Does the KDE uninstallation happen by itself without the need for leapp to initiate it?

Reply:
No, it is assumed leapp will initiate it.

Reply:
In that case, I would say that it should be part of this PR to tell leapp that the KDE pkgs are to be removed.
You would do that by producing `RpmTransactionTasks(to_remove=['kdepkg1', 'kdepkg2'])` in this actor.

@pirat89, @drehak, I fear that what I suggest above could cause a conflict in the transaction similar to https://github.com/oamg/leapp-repository/issues/177. But since the PESEventsScanner runs in the Facts phase, I see no way to deal with that. WDYT?

Reply:
As @pirat89 mentions, the PES data should actually take care of the removal, so disregard my point. I'll test that.